pkg/infra/kubernetes/manifests/deployment.yaml.tmpl

```diff
@@ -19,6 +19,7 @@ spec:
     metadata:
       labels:
         execUnit: {{ .ExecUnitName }}
+        elbv2.k8s.aws/pod-readiness-gate-inject: enabled
```

Comment:
we wont want this to always be injected because it really is only necessary for pods which are exposed.

I would make this be written only if the pod is exposed and also set in the aws provider since this is aws specific. If its too much to make this be set in the aws provider its ok to accept that as tech debt for now

Reply:
I was wondering about this. I can take a closer look.